tutorial/real_proj/github-org-analyzer.md

```diff
@@ -0,0 +1,244 @@
+
+* `to_string` : We use this function to convert integers to their string representations
+
+### **Step 6: Create the `print_org_repositories` Subroutine**
+
+Now let's write the `print_org_repositories` subroutine, which fetches and prints all the repositories of the "fortran-lang" organization using the GitHub API. This subroutine utilizes the `http-client` and `json-fortran` libraries to make API requests and handle JSON responses.
+
+1. Open the `github_org_analyzer.f90` file in the `src` folder.
+
+2. Create the `print_org_repositories` subroutine within the `github_org_analyzer` module:
+
+```fortran
+module github_org_analyzer
+   use json_module, only : json_file
+   use http, only : request, response_type
+   use stdlib_strings, only : to_string
+
+   implicit none
+   private
+
+   ! Declare the function as public to make it accessible to other modules
+   public :: print_org_repositories
+contains
+   subroutine print_org_repositories()
+      !! subroutine to print all repositories of fortran-lang organization
+      character(*), parameter :: api_url = 'https://api.github.com/orgs/fortran-lang/repos'
+      !! Stores GitHub API URL for fetching repositories
+      integer :: i
+      !! counter to traverse all the repos return by api
+      character(:), allocatable :: count
+      !! stores the string equivalent of counter, i.e variable i
+      character(:), allocatable :: value
+      !! stores the individual github repo name for each traversal
+      type(json_file) :: json
+      !! give the ability to parse the json content
+      type(response_type) :: response
+      !! stores the response from the github api
+      logical :: found
+      !!  flag for weather the current repo found or not
+
+      ! Make an HTTP GET request to the API URL and store the response in the `response` variable
+      response = request(url=api_url)
+
+      ! Checking for any errors. If an error occurs during the HTTP call, the `ok`
+      ! attribute will be set to .false., and the `err_msg` attribute will contain
+      ! the reason for the error.
+      if(.not. response%ok) then
+         print *, 'Request Fail: ', response%err_msg
+      else
+         print *, 'Fortran lang All repositories:'
+
+         ! Initialize the `json` object to parse the JSON content
+         call json%initialize()
+
+         ! Deserialize the JSON response(parsing the json)
+         call json%deserialize(response%content)
+
+         ! Traverse Repositories and Print Names
+
+         ! Counter to traverse all repos one by one
+         i = 1
+
+         ! Storing the string equivalent of i in the count variable
+         count = to_string(i)
+
+         ! Fetching the name of the 1st GitHub repository, if it exists (found is set to true)
+         call json%get('['//count//'].name', value, found)
+
+         ! Enter the loop to traverse all repositories while they exist
+         do while(found)
```

Comment:
This loop is strange. Does it call twice `json%get` for the same count? Why is the last `json%get` needed?